### ISSUE TITLE
conn.bind() failed when SASL GSSAPI auth is used and raise_exceptions=True

### ISSUE THREAD

**Issue by styleex** [cannatag/ldap3#224]
MVP:

```
import ldap3

s = ldap3.Server(fqdn, connect_timeout=5)
conn = ldap3.Connection(s, authentication=ldap3.SASL, sasl_mechanism=ldap3.GSSAPI, raise_exceptions=True)

# When used NTLM auth conn.bind() works well
# conn = ldap3.Connection(s, user='admin', password='secret', raise_exceptions=True, authentication=ldap3.NTLM)
conn.bind()

conn.search(search_base, search_filter, attributes=['cn', 'objectGUID', 'primaryGroupToken'])
```

raises exception:

```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    c.bind()
  File "/usr/lib/python3.5/dist-packages/ldap3/core/connection.py", line 488, in bind
    response = self.do_sasl_bind(controls)
  File "/usr/lib/python3.5/dist-packages/ldap3/core/connection.py", line 1049, in do_sasl_bind
    result = sasl_gssapi(self, controls)
  File "/usr/lib/python3.5/dist-packages/ldap3/protocol/sasl/kerberos.py", line 69, in sasl_gssapi
    result = send_sasl_negotiation(connection, controls, out_token)
  File "/usr/lib/python3.5/dist-packages/ldap3/protocol/sasl/sasl.py", line 159, in send_sasl_negotiation
    response = connection.post_send_single_response(connection.send('bindRequest', request, controls))
  File "/usr/lib/python3.5/dist-packages/ldap3/strategy/sync.py", line 124, in post_send_single_response
    responses, result = self.get_response(message_id)
  File "/usr/lib/python3.5/dist-packages/ldap3/strategy/base.py", line 364, in get_response
    raise LDAPOperationResult(result=result['result'], description=result['description'], dn=result['dn'], message=result['message'], response_type=result['type'])
ldap3.core.exceptions.LDAPSASLBindInProgressResult: LDAPSASLBindInProgressResult - 14 - saslBindInProgress - None - None - bindResponse - None
```

**Comment by cannatag** [comment]
Hi Styleex. Thanks for your support. I'm about to release version 2 of ldap3. Will check your issues against that version.

**Comment by cannatag** [comment]
SASLBindInProgress doesn't raise an exception anymore with raise_exceptions=True.
You can check code in dev. Will be released in version 2.1.

**Comment by cannatag** [comment]
2.1.0 released
